pkg/api/types.go

```diff
@@ -673,6 +673,9 @@ type GlusterfsVolumeSource struct {
 	// Required: EndpointsName is the endpoint name that details Glusterfs topology
 	EndpointsName string `json:"endpoints"`
 
+	// Optional: Servers is the gluster server list
+	EndpointsNameSpace *string `json:"endpointsns,omitempty"`
```

Comment:
Based on https://github.com/openshift/origin/issues/6331#issuecomment-248372663 it looks like this would allow a user in one namespace to cause a service and endpoints with names of their choice to be created in another namespace. That's a significant privilege escalation.